src/main/java/graphql/schema/idl/SchemaGeneratorHelper.java

```diff
@@ -1055,6 +1057,11 @@ Set<GraphQLDirective> buildAdditionalDirectiveDefinitions(BuildContext buildCtx)
         TypeDefinitionRegistry typeRegistry = buildCtx.getTypeRegistry();
 
         for (DirectiveDefinition directiveDefinition : typeRegistry.getDirectiveDefinitions().values()) {
+            if (IncludeDirective.getName().equals(directiveDefinition.getName())
```

Comment:
we should add all the system defined directives

@deprecated / @specfiedBy / @include / @skip

Reply:
Otherwise I approve of this PR and its approach

Reply:
It's a little different for `@deprecated` and `@specifiedBy`. 
Let me explain and tell me what you think. 

`@deprecated` and `@specifiedBy` are schema directives, and the current behavior is to [only add them if not present](https://github.com/tinnou/graphql-java/blob/93d7f5df0db26f4e55506a97d69c7ce537e8638e/src/main/java/graphql/schema/idl/SchemaGenerator.java#L78) in the supplied SDL. 
I think this is fair since one could imagine a case where we are reading an older spec SDL and it would be nice if we can keep the spec behavior defined in the SDL (e.g deprecated locations missing the enum location on older version of the spec) although not super important in my opinion since spec evolution is backward compatible, hence using the engine definitions is still OK.

Additionally `@deprecated` and `@specifiedBy` are added in the typeRegistry rather than the GraphQLSchema.Builder since `@deprecated` and `@specifiedBy` are schema directives, and the applied directives locations should be checked while performing the registry checks.

Also we could add some validation, such that if you attempt to define `@deprecated` and `@specifiedBy` on your own, then they need to be "compatible" with the spec standard defined directives. Not super convinced we need that kind of validation, it feels a little over defensive.

For `@skip and `@include`, they are execution (query) directives, i.e there can't be any applied directives in the schema, so they are being added by default only later in the GraphQLSchema builder. The behavior I added here "ignores" any directive in the SDL that attempts to redefines them. 

To make things consistent w.r.t to all standard directives, we could follow the same pattern as with deprecated and specifiedBy, meaning adding them all only if not present, but we would need to either remove them from the GraphQLSchema.Builder (potentially a behavior breaking change) or only add them to the GraphQLSchem.Builder if not present.

Thoughts?

Reply:
Lets go with this for now - progress over perfection.  You make a good point so lets go with it